### PR TITLE
Fix flaky tests

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/tests/Diagnostics/DiagnosticScopeLiveTests.cs [hunk 1]
```diff
@@ -16,14 +16,27 @@ namespace Azure.Messaging.ServiceBus.Tests.Diagnostics
...
+        [SetUp]
+        public void Setup()
+        {
+            _listener = new TestDiagnosticListener(EntityScopeFactory.DiagnosticNamespace);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            _listener.Dispose();
```
Comment (JoshLove-msft) [inline sdk/servicebus/Azure.Messaging.ServiceBus/tests/Diagnostics/DiagnosticScopeLiveTests.cs:30]: These tests occasionally have errors that indicate they are stomping on each others events, but I'm not sure how that is possible given that these are being run in NonParallelizable mode. Trying explicit teardown to see if it helps.

Comment (jsquire) [reply sdk/servicebus/Azure.Messaging.ServiceBus/tests/Diagnostics/DiagnosticScopeLiveTests.cs:30]: yay, ambient diagnostic state!

**Review by jsquire — APPROVED** [review]

**Review by MiYanni — APPROVED** [review]

**Pull request merged**